cmd/searcher/search/search_regex.go

```diff
@@ -411,13 +410,14 @@ func regexSearch(ctx context.Context, rg *readerGrep, zf *store.ZipFile, fileMat
 						fm.Path = f.Name
 					}
 				}
-				if match {
+
+				if match && !isPatternNegated || !match && isPatternNegated {
```

Comment:
nit: Is this correct? I guess the tests pass. But I normally put parenthesis in to make it clear. This could also just be simplified to:

```suggestion
				if match == !isPatternNegated {
```

Reply:
followed your suggestion